internal/pkg/scaffold/olm-catalog/csv.go

```diff
@@ -188,29 +189,42 @@ func (s *CSV) getCSVPath(ver string) string {
 	return filepath.Join(s.pathPrefix, OLMCatalogDir, lowerProjName, ver, name)
 }
 
-// initCSVFields initializes all csv fields that should be populated by a user
-// with sane defaults. initCSVFields should only be called for new csv's.
-func (s *CSV) initCSVFields(csv *olmapiv1alpha1.ClusterServiceVersion) {
-	// Metadata
+// setCSVDefaultFields sets all csv fields that should be populated by a user
+// to sane defaults.
+func (s *CSV) setCSVDefaultFields(csv *olmapiv1alpha1.ClusterServiceVersion) {
+	// These fields have well-defined required values.
 	csv.TypeMeta.APIVersion = olmapiv1alpha1.ClusterServiceVersionAPIVersion
 	csv.TypeMeta.Kind = olmapiv1alpha1.ClusterServiceVersionKind
 	csv.SetName(getCSVName(strings.ToLower(s.OperatorName), s.CSVVersion))
-	csv.SetNamespace("placeholder")
-	csv.SetAnnotations(map[string]string{"capabilities": "Basic Install"})
-
-	// Spec fields
-	csv.Spec.Version = *semver.New(s.CSVVersion)
-	csv.Spec.DisplayName = k8sutil.GetDisplayName(s.OperatorName)
-	csv.Spec.Description = "Placeholder description"
-	csv.Spec.Maturity = "alpha"
-	csv.Spec.Provider = olmapiv1alpha1.AppLink{}
-	csv.Spec.Maintainers = make([]olmapiv1alpha1.Maintainer, 0)
-	csv.Spec.Links = make([]olmapiv1alpha1.AppLink, 0)
-}
 
-// setCSVDefaultFields sets default fields on older CSV versions or newly
-// initialized CSV's.
-func setCSVDefaultFields(csv *olmapiv1alpha1.ClusterServiceVersion) {
+	// Set if empty.
+	if csv.GetNamespace() == "" {
+		csv.SetNamespace("placeholder")
+	}
+	if csv.GetAnnotations() == nil {
+		csv.SetAnnotations(map[string]string{})
+	}
+	if caps, ok := csv.GetAnnotations()["capabilities"]; !ok || caps == "" {
+		csv.GetAnnotations()["capabilities"] = "Basic Install"
+	}
+	if csv.Spec.Provider == (olmapiv1alpha1.AppLink{}) {
```

Comment:
Nit: are the parens around `olmapiv1alpah1.AppLink{}` necessary?

Reply:
They are when [comparing](https://stackoverflow.com/a/28447372) struct types directly.

Reply:
Ahh, TIL!